functions-and-scope.js

```diff
@@ -53,25 +84,55 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // averageGrade([6, 4, 5]) geeft xxxx
 // averageGrade([8, 9, 4, 6, 10]) geeft xxxx
 
+function averageGrade(arrayGrades) {
+    let sumGrades = 0;
+    for (let i = 0; i < arrayGrades.length ; i++) {
+        sumGrades += arrayGrades[i];
+    }
+    return sumGrades / arrayGrades.length;
+}
+
+console.log("Opdracht 2b")
+console.log(averageGrade(grades));
+console.log(averageGrade([6, 4, 5]));
+console.log(averageGrade([8, 9, 4, 6, 10]));
+
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+//Ik wilde dit eerst met Math.round doen, maar dit geeft geen twee decimalen.
+//console.log(Math.round(averageGrade(grades)));
 
+console.log("Opdracht 2c")
+console.log(averageGrade(grades).toFixed(2));
+console.log(averageGrade([6, 4, 5]).toFixed(2));
+console.log(averageGrade([8, 9, 4, 6, 10]).toFixed(2));
 
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
 // Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
-// * Hoe kan ik iedere waarde van de array langsgaan?
-// * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// * Hoe kan ik iedere waarde van de array langsgaan? - for loop
+// * Op welke conditie moet ik checken? - is dit cijfer hoger dan het hoogste cijfer dat ik al heb, een variabele hiervoor aanmaken
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan? - de variabele die voor het hoogste cijfer is gemaakt gebruiken
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomst: 9
 
+let highGrade = 0;
```

Comment:
Het is geen goed voorbeeld, maar ik wil het je wel meegeven:
Wanneer je een array meekrijgt ([-1, -3, -2]) met alleen maar negatieve getallen, dan zal de functie altijd 0 teruggeven. Het is daarom verstandig om altijd de eerste waarde in de array te pakken:
```suggestion
let highGrade = grades[0];
```